bn-apps/cordapp-updates-distribution/README.md

```diff
@@ -0,0 +1,270 @@
+7. Schedule periodic synchronization by invoking [ScheduleSyncFlow](#scheduling-synchronisation) from Corda shell on the nodes of participants.
+8. Done. When published, updates will appear in the local repositories of participants. Please bear in mind that node administrators will still have to [install updates manually](https://docs.corda.net/releases/release-M8.2/creating-a-cordapp.html#installing-apps).
+
+Please read through the rest of the document for more information about operational considerations and available configuration options.
+ 
+# Structure
+
+*CorDapp Distribution Service* consists of the following components:
+* **corda-updates-core** - API over Maven Resolver
+* **corda-updates-transport** - custom transport implementations for Maven Resolver over Corda Flows
+* **corda-updates-app** - CorDapp that allows participants to schedule periodic updates and provides basic version reporting functionality.
+* **corda-updates-shell** - command line interface 
+
+# corda-updates-core
+
+`corda-updates-core` sits in the heart of CorDapp Distribution Service and provides APIs on top of Maven Resolver to download CorDapps from remote Maven repositories over `file`, `http(s)`, `corda-flows` and `corda-rpc` transports. `corda-updates-core` is used internally only and is not visible to the end users. It effectively allows to:
+* Get a list of artifact versions available in a remote repository
+* Download a single version of an artifact from a remote repository
+* Download a version range from a remote repository, which is effectively combination of the two previous steps.
+* Associate different CorDapps with different remote repositories and to sync them down via a single method invocation. 
+
+`corda-updates-core` also supports basic authentication for HTTP proxies and remote repositories. 
+
+# corda-updates-transport
+
+`corda-updates-transport` provides a bespoke Maven Resolver transports over Corda Flows and Corda RPC.
+
+## Why do we need it?
+
+Corda-based transports allow repository hosters to enforce their custom rules onto incoming requests and to filter out any unauthorised download attempts. For example CorDapp Distribution Service can be easily integrated with [Business Networks Membership Service](https://github.com/corda/corda-solutions/tree/master/bn-apps/memberships-management), that would effectively allow it to filter out any non Business Network traffic. 
+
+## Session Filters
+
+Corda-based transports allow developers to implement their custom `SessionFilter`s that can be used to reject any unintended download requests. `SessionFilter` is a simple interface that, if implemented, is invoked against every incoming download request. `SessionFilter` returns a boolean that indicates whether the request should be let through or not.
+
+```kotlin
+interface SessionFilter {
```

Comment:
This looks pretty reasonable. But I wonder if the right way to do this is use the support for sub-classing flows that @roastario has worked on.

Reply:
Should we support both? For backwards compatibility

Reply:
Does the app need to support Corda 3? If you can target Corda 4 there's no need as the ability to override flows is in review right now.

Reply:
Yes, makes sense. No need to support both I think. I'll change ^^ logic once Stefano's PR is merged.